### PR TITLE
[wip] perf(runtime): use raw v8 slots

### PR DESCRIPTION
**Pull request by littledivy**
Depends on https://github.com/denoland/rusty_v8/pull/1092

rusty_v8 Annex stuff is slow.

### CODE REVIEW

**Comment by andreubotella** [comment]
Maybe I'm missing something, but wouldn't this result in the values in the slots leaking, since dropping the isolate doesn't drop the slots?

File: core/runtime.rs [hunk 1]
```diff
@@ -548,9 +558,9 @@ impl JsRuntime {
     isolate.set_host_initialize_import_meta_object_callback(
       bindings::host_initialize_import_meta_object_callback,
     );
-    isolate.set_host_import_module_dynamically_callback(
-      bindings::host_import_module_dynamically_callback,
-    );
+    // isolate.set_host_import_module_dynamically_callback(
+    //   bindings::host_import_module_dynamically_callback,
+    // );
```
Comment (andreubotella) [inline core/runtime.rs:563]: Is there any reason for commenting this out?

**Comment by andreubotella** [comment]
The `Rc` has to be dropped at some point after the context is GC'd. But at that point you don't have access to the slots, so you'd probably need something like:

```rust
let context_ptr = Rc::into_raw(Rc::<RefCell<ContextState>>::default()) as *mut c_void;
unsafe {
  context.open(&mut isolate).set_aligned_pointer_in_embedder_data(
    CONTEXT_STATE_SLOT,
    context_ptr,
  )
};
let weak = v8::Weak::with_guaranteed_finalizer(&mut isolate, context, move || {
  let _ = unsafe { Rc::from_raw(context_ptr as *mut ContextState) };
});
store_somewhere_for_the_duration_of_the_isolate(weak);
```

File: core/runtime.rs (hunk 1 above)
Comment (bartlomieju) [reply core/runtime.rs:563]: It's by mistake. I'll revert it

**Comment by bartlomieju** [comment]
@littledivy do we still want to land this PR?

**Pull request closed without merging**